### ISSUE TITLE
Select2Widget ignores empty_label 

### ISSUE THREAD

**Issue by dyve** [applegrew/django-select2#565]
**Describe the bug**
When using `Select2Widget` on a field that has `empty_label` set, the `empty_label` setting is ignored. I would expect the content of `empty_label` (if any) as the default value for `data-placeholder` in the `attrs` of `select2Widget`.

**Code Snippet**
```
class BookForm(forms.ModelForm):
    """Model Book has field user that is ForeignKey to User."""
    class Meta:
        model = Book
        fields = ["user"]
        widgets = {"user": Select2Widget}

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fields["user"].empty_label = "No user is fine too."
        # self.fields["user"].widget.attrs['data-placeholder'] = self.fields["user"].empty_label
        # Uncomment the line above to make the empty_label appear as the placeholder
```

**To Reproduce**
1. Use the form from the example.
2. See that "No user is fine too." does NOT appear.
3. Uncomment the line that enables the workaround.
4. See that "No user is fine too." DOES appear.

**Expected behavior**
When a field has an `empty_label`, I expect that to be the default value for Select2Widget.attrs["data-placeholder"]

**Comment by codingjoe** [comment]
Hi @dyve very good catch, this seems to be a bug in deed. I believe the problem is here:
https://github.com/applegrew/django-select2/blob/8bd7f7209f29cdeffc9894a138c2e815c50b75b9/django_select2/forms.py#L80
as well as here:
https://github.com/applegrew/django-select2/blob/8bd7f7209f29cdeffc9894a138c2e815c50b75b9/django_select2/forms.py#L94
here:
https://github.com/applegrew/django-select2/blob/8bd7f7209f29cdeffc9894a138c2e815c50b75b9/django_select2/forms.py#L449
and here:
https://github.com/applegrew/django-select2/blob/8bd7f7209f29cdeffc9894a138c2e815c50b75b9/django_select2/forms.py#L404

The solution is simple. Instead of using an empty string, this should point to `self.empty_label`.
Would you be interested in providing a pull-request that patches the issue? I would really welcome the support.

Best
-Joe

**Comment by codingjoe** [comment]
I went ahead and wrote a patch myself, to make sure this feature reaches users asap. Thanks!

**Comment by codingjoe** [comment]
Released in 7.1.1, thx!

**Comment by dyve** [comment]
Thanks @codingjoe!
